lms/util/lti_launch.py

```diff
@@ -11,55 +11,46 @@ def get_application_instance(session, consumer_key):
     ).one()
 
 
-def default_get_secret(request, consumer_key):
+def get_secret(request, consumer_key):
     """Retrieve the lti secret given."""
     instance = get_application_instance(request.db, consumer_key)
     return instance.shared_secret
 
 
-def default_get_lti_launch_params(request):
+def get_lti_launch_params(request):
```

Comment:
One of the things I initially found confusing about this codebase was that it sometimes referred to the entire set of `request.params` as `lti_launch_params`, when in fact non-LTI things could absolutely be mixed in. Anyway...that's an aside...

Reply:
I haven't looked into that yet. There's a few different views where LTI launch params are received and should be validated (but never are) so I wonder if in the long run we want a validation schema for validating LTI launch request params, perhaps with sub-schemas for different kinds of launches (assignment launch, content item selection launch, ...)? And then any post-validation code can safely assume that the necessary params will be present.

Are there sometimes non-LTI params mixed in with LTI launch request params in the request params? I haven't spotted that yet